Sidecar message consumption AVRO deserialization assumes records

Avro can serialize lots of things that are not records (primitives, etc.), but[ RecordDeserializer.java::handleAvro() ](https://github.com/confluentinc/ide-sidecar/blob/main/src/main/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializer.java#L134) assumes Record, and croaks when deserializing, say, a key schema describing "just a long encoding of a timestatmp", used as key schema of topic WeatherData within our test custom-data-cluster.

Schema WeatherData-key:
```
{
  "logicalType": "timestamp-micros",
  "type": "long"
}
```

The exception raised:

....
Caused by: java.lang.ClassCastException: java.lang.Long cannot be cast to org.apache.avro.generic.GenericData$Record
	at io.confluent.idesidecar.restapi.messageviewer.RecordDeserializer.handleAvro(RecordDeserializer.java:134)
...

--- a/src/main/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializer.java
+++ b/src/main/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializer.java
@@ -46,6 +46,7 @@ import java.time.format.DateTimeFormatter;
 import java.util.Map;
 import java.util.Optional;
 import java.util.function.Function;
+
 import org.apache.avro.generic.GenericData;
 import org.apache.avro.generic.GenericDatumWriter;
 import org.apache.avro.io.EncoderFactory;
@@ -131,17 +132,20 @@ public class RecordDeserializer {
         var avroDeserializer = new KafkaAvroDeserializer(sr);
     ) {
       avroDeserializer.configure(SERDE_CONFIGS, isKey);
-      var avroRecord = (GenericData.Record) avroDeserializer.deserialize(topicName, bytes);
-      var writer = new GenericDatumWriter<>(
-          avroRecord.getSchema());
-      var encoder = EncoderFactory.get().binaryEncoder(outputStream, null);
-      writer.write(avroRecord, encoder);
-      encoder.flush();
-      var jacksonAvroSchema = new AvroSchema(avroRecord.getSchema());
-      return AVRO_OBJECT_MAPPER
-          .readerFor(ObjectNode.class)
-          .with(jacksonAvroSchema)
-          .readValue(outputStream.toByteArray());
+      var genericObject = avroDeserializer.deserialize(topicName, bytes);
+      if (genericObject instanceof GenericData.Record avroRecord) {
+        var writer = new GenericDatumWriter<>(avroRecord.getSchema());
+        var encoder = EncoderFactory.get().binaryEncoder(outputStream, null);
+        writer.write(avroRecord, encoder);
+        encoder.flush();
+        var jacksonAvroSchema = new AvroSchema(avroRecord.getSchema());
+        return AVRO_OBJECT_MAPPER
+            .readerFor(ObjectNode.class)
+            .with(jacksonAvroSchema)
+            .readValue(outputStream.toByteArray());
+      } else {
+        return OBJECT_MAPPER.valueToTree(genericObject);
+      }
     } catch (IOException e) {
       throw new RuntimeException("Failed to deserialize Avro-encoded bytes", e);
     }

--- a/src/main/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializer.java
+++ b/src/main/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializer.java
@@ -46,7 +46,6 @@ import java.time.format.DateTimeFormatter;
 import java.util.Map;
 import java.util.Optional;
 import java.util.function.Function;
-
 import org.apache.avro.generic.GenericData;
 import org.apache.avro.generic.GenericDatumWriter;
 import org.apache.avro.io.EncoderFactory;

